src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -32621,21 +32731,21 @@ void M7([NotNull]string x)
                 // (7,9): warning CS8602: Dereference of a possibly null reference.
                 //         x.ToString(); // 1
                 Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x").WithLocation(7, 9),
-                // (24,13): warning CS8625: Cannot convert null literal to non-nullable reference type.
-                //         x = null; // 2
-                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(24, 13),
-                // (25,19): warning CS8600: Converting null literal or possible null value to non-nullable type.
-                //         (x, _) = (null, 1); // 3
-                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "null").WithLocation(25, 19),
+                // (26,5): warning CS8777: Parameter 'x' must have a non-null value when exiting.
+                //     } // 2
+                Diagnostic(ErrorCode.WRN_ParameterDisallowsNull, "}").WithArguments("x").WithLocation(26, 5),
                 // (35,16): warning CS8603: Possible null reference return.
-                //         return null; // 4
+                //         return null; // 3
                 Diagnostic(ErrorCode.WRN_NullReferenceReturn, "null").WithLocation(35, 16),
                 // (40,13): warning CS8600: Converting null literal or possible null value to non-nullable type.
-                //         x = null; // 5
+                //         x = null; // 4
                 Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "null").WithLocation(40, 13),
                 // (41,19): warning CS8600: Converting null literal or possible null value to non-nullable type.
-                //         (x, _) = (null, 1); // 6
-                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "null").WithLocation(41, 19)
+                //         (x, _) = (null, 1); // 5
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "null").WithLocation(41, 19),
+                // (42,5): warning CS8777: Parameter 'x' must have a non-null value when exiting.
+                //     } // 6
```

Comment:
Why enforce this on parameters with RefKind.None? Isn't the method just asserting about the value originally passed as an argument in that case?

On the other hand, it doesn't feel like there's a use case for assigning maybe-null to a NotNull, non-ref parameter, so it probably isn't very important to remove this warning.

Reply:
Thinking about it for a moment, it seems like unnecessary effort to attempt to remove this warning--but it might not be what we want to *prevent* this warning by just assigning not-null to the parameter. e.g.

```cs
void MyAssert1([NotNull] object? obj)
{
    // hooray, we know the argument passed by the caller is non-null after the method returns
    if (obj is null)
        throw new ArgumentNullException(nameof(obj));
}

void MyAssert2([NotNull] object? obj)
{
    obj = new object(); // we fixed the warning, hooray!
    // but caller still might be holding a null..
}
```

Reply:
Good point. I don't have a strategy to distinguish between those two cases :-(